test/utils/describeConformanceV5.js

```diff
@@ -57,7 +57,38 @@ function testThemeComponents(element, getOptions) {
       expect(container.firstChild).to.have.attribute(testProp, 'testProp');
     });
 
-    it("respect theme's styleOverrides", () => {
+    it("respect theme's styleOverrides custom state", () => {
+      const { muiName, testStateOverrides } = getOptions();
+
+      if (!testStateOverrides) {
+        return;
+      }
```

Comment:
When should this not be tested?

Reply:
If you ask me, I would say most of the time we don't need to test it. We don't even fully test it right now, we take an arbitrary case. I would personally prefer we use the Button as a stress test template. 

> Badge is still untested.

The badge doesn't have state on it's root element.

Reply:
I have moved away from the API in #24095 with this patch https://gist.github.com/oliviertassinari/60614c8c37d4ba10adc2a3b1aba1277e. The fix is meant for backward compatibility with v4.

Reply:
> >Badge is still untested.

>The badge doesn't have state on its root element.

Then why did we need to change the implementation?

> If you ask me, I would say most of the time we don't need to test it. 

It already broke once.

> The fix is meant for backward compatibility with v4.

`describeConformanceV5` is not meant for v4.

Reply:
> It already broke once.

Exactly why I suggest the current tradeoff. If it breaks a second time, it will be a signal that it isn't enough.

Reply:
> If it breaks a second time, it will be a signal that it isn't enough.

What made you change your opinion? How did you decide that one breakage is ok but not two?

Reply:
> What made you change your opinion?

The current test tradeoff is a strength of the initial proposal, I'm personally happy with it.

> How did you decide that one breakage is ok but not two?

I don't think that it's about the number of times it breaks. I think that we can break something 10 times as long as we learn something each time, we move forward, not backward. I'm proposing a greedy tradeoff. If we can get away with it, if it's enough, then that the best possible outcome.

> How do you suppose do we find it breaks a second time?

We can expect a developer to report it, like this time. If nobody reports it's likely safe to sit on it.

Reply:
> We can expect a developer to report it, like this time. If nobody reports it's likely safe to sit on it.

But this didn't happen this time? What changed between report 1 and report 2?

> I don't think that it's about the number of times it breaks. I think that we can break something 10 times as long as we learn something each time, that we move forward, not backward, that we don't make the same mistake each time.

You did say that it is about the number. How do you think we make sure it doesn't break again when we didn't add a test?

> The current test tradeoff is a strength of the initial proposal, I'm personally OK with it.

What does this even mean?

Reply:
> But this didn't happen this time? What changed between report 1 and a future report 2?

Two thoughts. 1. If a wrong behavior isn't reported and goes unnoticed for a long time, it's probably not important, hence so *should likely*/*can* be ignored. 2. This regression got reported in #24262 after I opened this PR.

> How do you think we make sure it doesn't break again when we didn't add a test?

I have added new test cases, hopefully enough.

> What does this even mean?

I would personally only add a single test case for the slider or the button. But based on your feedback and Marija's one, I added a new entry in the describe conformance. I think that the test tradeoff is oversized to the problem, but I'm happy to give it a try.

Reply:
I still don't understand when this doesn't need to be tested. The Badge implementation changed but no test was added.

Reply:
> The Badge implementation changed but no test was added.

The argument I have tried to defend is that it doesn't matter. I think that 80% of the value and what needs to be tested is the approach itself, testing a few components is enough.

If we break it later down the road, then it's a signal it's not enough, that we need to test more. I defend that we should be lazy.

On a different note, the Badge is different, there is no state on the root component, the new test inside `describeConformanceV5` doesn't support it.

Reply:
> The argument I have tried to defend is that it doesn't matter. 

It does matter. Every implementation needs a test unless it isn't testable at which point one needs to make an argument why it isn't testable. 

We don't decide by who writes the code if it needs to be tested.

Reply:
> very implementation needs a test

I respectfully disagree

Reply:
> I respectfully disagree. I think that we should also add "unless it's not worth it".

Impossible to collaborate on that premise since this argument applies anytime.

> I think that it's like the 100% code coverage I have tried to push in the past and we gave up on. 

At some point there comes a time where you should educate yourself more on the topic. I can only bring nudge you so far but if you believe that line coverage has anything to do with behavior testing I don't think you put any thought into verifying your work. 

It becomes evident to me that you don't know at all why you made this change. I'm asking you a third time now: Why was the Badge changed but no test added? If you don't know what the code does then please revert it.

Reply:
> Why was the Badge changed but no test added?

I have tried to answer it in https://github.com/mui-org/material-ui/pull/24253#discussion_r553239987, the new generic test inside describeConformanceV5 doesn't support it. It's a rounding error.

Reply:
I'm sorry but you have the responsibility to explain the code you write. We have this expectation for any other contribution. What you did, expected behavior, actual behavior applies to everyone.